routers/web/base.go

```diff
@@ -37,13 +37,15 @@ func storageHandler(storageSetting setting.Storage, prefix string, objStore stor
 					return
 				}
 
-				if !strings.HasPrefix(req.URL.RequestURI(), "/"+prefix) {
+				if !strings.HasPrefix(req.URL.EscapedPath(), "/"+prefix+"/") {
```

Comment:
It seems that here should be `if !strings.HasPrefix(req.URL.Path, "/"+prefix+"/") {`
because below there is `rPath := strings.TrimPrefix(req.URL.Path, "/"+prefix+"/")`.

And another code block is also using it for L76

Reply:
yup I agree

```suggestion
				if !strings.HasPrefix(req.URL.Path, "/"+prefix+"/") {
```